tests/python/unittest/test_arith_solve_linear_inequality.py

```diff
@@ -0,0 +1,187 @@
+    assert ir.structural_equal(solution[0], x >= (y + 10))
+    assert ir.structural_equal(solution[1], x <= (20 - y))
+    assert ir.structural_equal(solution[2], y >= 0)
+    assert ir.structural_equal(solution[3], y <= 5)
+
+    # solve and get the ranges
+    solution = arith.solve_linear_inequalities([
+        tvm.tir.LE(x + y, 20),
+        tvm.tir.GE(x - y, 10),
+    ], [x, y], ranges)
+    # 0 <= y <=5
+    assert solution.ranges[y].min == 0
+    assert solution.ranges[y].extent == 6
+    # y + 10 <= x <= 20 - y
+    assert ir.structural_equal(solution.ranges[x].min, y + 10)
+    assert solution.ranges[x].extent == 11  # max(10 - 2y)
+
+    # deskew the solved ranges to be starting from zero
+    solution = arith.solve_linear_inequalities(problem, variables, ranges, deskew_range=True)
+    [x_new, y_new] = solution.dst.variables
+    [rel] = solution.dst.relations
+    assert ir.structural_equal(rel, (y_new*2) + x_new <= 10)
+    assert ir.structural_equal(solution.dst.ranges[x_new].min, 0)
+    assert ir.structural_equal(solution.dst.ranges[x_new].extent, 11)
+    assert ir.structural_equal(solution.dst.ranges[y_new].min, 0)
+    assert ir.structural_equal(solution.dst.ranges[y_new].extent, 6)
+    assert ir.structural_equal(solution.src_to_dst[x], x_new + (y_new + 10))
+    assert ir.structural_equal(solution.src_to_dst[y], y_new)
+    assert ir.structural_equal(solution.dst_to_src[x_new], x - y - 10)
+    assert ir.structural_equal(solution.dst_to_src[y_new], y)
+
+
+def test_equal():
+    x, y = te.var("x"), te.var("y")
+    problem = [
+        tvm.tir.GE(x + y, 10),
+        tvm.tir.GE(x - y, 2),
+        tvm.tir.LE(x, 6),
+    ]
+
+    solution = arith.solve_linear_inequalities(problem, [x, y])
+    assert solution.ranges[x].min == 6
+    assert solution.ranges[x].extent == 1
+    assert solution.ranges[y].min == 4
+    assert solution.ranges[y].extent == 1
+
+    solution = arith.solve_linear_inequalities(problem, [x, y], deskew_range=True)
+    assert len(solution.dst.variables) == 0
+    assert len(solution.dst.ranges) == 0
+    assert len(solution.dst.relations) == 0
+    assert solution.src_to_dst[x] == 6
+    assert solution.src_to_dst[y] == 4
+
+
+def test_multi_equal():
+    x, y, z = te.var("x"), te.var("y"), te.var("z")
+    problem = [
+        tvm.tir.LE(x, 6),
+        tvm.tir.GE(x, 6),
+        tvm.tir.GE(x - z * y, 0),
+        tvm.tir.LE(x - z * y, 0),
+    ]
+
+    solution = arith.solve_linear_inequalities(problem, [x, y, z])
+    assert solution.ranges[x].min == 6
+    assert solution.ranges[x].extent == 1
+
+    solution = arith.solve_linear_inequalities(problem, [x, y, z], deskew_range=True)
+    assert solution.src_to_dst[y] == y
+    assert solution.src_to_dst[z] == z
+    assert solution.src_to_dst[x] == 6
+
+
+def test_no_solution():
+    x = te.var("x0")
+    vranges = {x: tvm.ir.Range.make_by_min_extent(-20, 41)}
+    problem = [-x - 4 <= -5*x + 2, x*4 + 5 <= x*5]
```

Comment:
No solution with more variables can be added.

Reply:
more cases are covered in `test_solution_consistency`